src/main.rs

```diff
@@ -1,14 +1,19 @@
 //! Compute@Edge static content starter kit program.
 
-mod awsv4;
 mod config;
 
-use crate::awsv4::hash;
-use chrono::Utc;
-use fastly::handle::dictionary::DictionaryHandle;
 use fastly::http::{header, HeaderValue, Method, StatusCode};
 use fastly::{Error, Request, Response};
 
+#[cfg(feature = "auth")]
+mod awsv4;
+
+#[cfg(feature = "auth")]
+use crate::awsv4::{hash, Utc};
+
+#[cfg(feature = "auth")]
+use fastly::handle::dictionary::DictionaryHandle;
```

Comment:
I _think_ this might be a good place to use the `cfg-if` crate. (See: https://crates.io/crates/cfg-if)

Unless I'm mistaken, and apologies if that's the case, we could use that to write something like

```rust
cfg_if::cfg_if! {
    if #[cfg(feature = "auth")] {
        use foo;
        use bar;
        use baz;
    }
}
```